apis/python/src/tiledbsoma/common_nd_array.py

```diff
@@ -38,8 +38,12 @@ def create(
         :param type: The Arrow type to be stored in the NDArray.
             If the type is unsupported, an error will be raised.
 
-        :param shape: the length of each dimension as a sequence, e.g., ``[100, 10]``.
-            All lengths must be in the positive int64 range.
+        :param shape: The maximum capacity of the dataframe, including room for any
+            intended future appends,as a sequence. E.g. ``[100, 10]``.  All
+            lengths must be in the positive int64 range, or `None`.  If a slot
+            is `None` -- only supported for `SparseNDArray`, not `DenseNDArray`
+            -- then the maximum possible size will be used. This makes a
```

Comment:
given that SOMA has a single dimension type for ndarray, and therefore a well-defined domain range, lets just say what the default size is going to be (ie.., the default, if None is specified, is a capacity of 2^64).

Reply:
@bkmartinjr I found via unit test that tiledb core requires that the capacity, rounded up to an exact multiple of the extent, needs to be representable as a signed 64-bit integer

so in particular when a unit test (or anyone else) sets extent to a not-power-of-two number like 999 or 1000 then the create fails

so i went with this wording instead since users _cannot_ always get 2**63

i can add a comment to that effect if that helps